Add profile onboarding mutators and selectors to wc-api

joshuatf opened the pull request:
Fixes #2303

Creates an easy way to store and retrieve the profiler information in the state via the API.

### Detailed test instructions:

1.  Set the `skipped` and `completed` properties to (boolean) false.
2. Note that the profiler is shown.
3. In an HOC add `const { updateProfileItems } = dispatch( 'wc-api' );` in any component.
4. Update the skipped property `this.props.updateProfileItems( { skipped: true } );`.
5. Note that the property is updated and the profiler is hidden.

justinshreve commented on includes/features/onboarding/class-wc-admin-onboarding.php:
```diff
@@ -41,28 +41,23 @@ public function __construct() {
 	 */
 	public function should_show_profiler() {
-		// @todo Remove this once we have a proper way to dismiss the profiler.
-		if ( ! defined( 'WOOCOMMERCE_ADMIN_DEV_SHOW_PROFILER' ) || false === WOOCOMMERCE_ADMIN_DEV_SHOW_PROFILER ) {
```
This was a short lived change 😆.

joshuatf replied:
Haha, I didn't realize it would be gone so quick either.

joshuatf commented on client/wc-api/onboarding/operations.js:
```diff
@@ -0,0 +1,97 @@
+ */
+import { getResourceName } from '../utils';
+
+const NAMESPACE = '/wc-admin/v1';
```
Good idea.  Added that and replaced references to `/wc-admin/v1` in https://github.com/woocommerce/woocommerce-admin/pull/2310/commits/6c2bb2f3b0f34158600769490d283e853129aad1

joshuatf commented:
Thanks for the review @justinshreve!  Implemented your feedback and replaced constants where applicable.

>Edit: Also needs the base branch updated against master.

Ha, thank you- feel like I needed this reminder today 😅
